drizzle-orm/src/mysql-core/dialect.ts

```diff
@@ -965,3 +977,20 @@ export class MySqlDialect {
 		};
 	}
 }
+
+// Had to reuse the constructor logic from `is` to avoid circular dependencies
+const isCustomColumn = (c: MySqlColumn): c is AnyMySqlCustomColumn => {
```

Comment:
What is the variable giving the circular dependency issue? (I guess it is `MySqlColumn`)

Maybe if you do the types refactor I mention in the other comment, the circular dependency gets solved, and this function can be removed.

Tiny thing: The comment was a bit confusing for me, specially the word "reuse", maybe I would change to "duplicate" or "copy".

Anyway, I suggest avoiding duplicating the code. I know you did it as last resource, but I think we'll be able to find a way around 😄, let's test the types refactor and see what happens.

Reply:
I would like to wait what Dan thinks about this because we've been having some problem with circular dependencies. I had to do something similar to this in the `sql` file.